### ISSUE TITLE
conda update --all gives different unsatisfiable package specifications on each run

### ISSUE THREAD

**Issue by bryanwweber** [conda/conda#1388]
Hello,
I'm using Anaconda 2.2.0 with Python 3.4 on Windows 7. I also have the `accelerate` and `iopro` packages installed. I'm trying to run `conda update --all`, and its failing with the "Unsatisfiable package specifications" error. However, every time I run the update command, it gives a different set of packages that are conflicting. In addition, if I run the `conda search --outdated` command, I can update each of the listed packages manually. Running `conda update --all --alt-hint` gives the following output:

```
Fetching package metadata: ....
Solving package specifications: ..
Error: Unsatisfiable package specifications.
Generating minimal hint:
[      COMPLETE      ]|##################################################| 100%

The following set of clauses is unsatisfiable:

python-2.7.3-6 => not python-3.4.2-0
python-2.6.9-1 => not python-3.4.2-1
scipy-0.12.0-np16py27_p0 => python-2.7.10-0 or python-2.7.3-2 or python-2.7.3-3 or python-2.7.3-4 or python-2.7.3-5 or p
ython-2.7.3-6 or python-2.7.3-7 or python-2.7.4-0 or python-2.7.5-0 or python-2.7.5-1 or python-2.7.5-2 or python-2.7.6-
0 or python-2.7.6-2 or python-2.7.7-0 or python-2.7.7-1 or python-2.7.7-2 or python-2.7.8-0 or python-2.7.9-0 or python-
2.7.9-1
mklfft-1.0-np19py34_p0 => mkl-rt-11.1-p0
python-2.6.8-5 => not python-2.7.4-0
scipy-0.12.0-np16py27_p0 => mkl-rt-11.0-p0 or mkl-rt-11.0-p1 or mkl-rt-11.0-p2
python-2.6.8-6 => not python-3.4.1-2
python-2.7.5-1 => not python-3.4.2-0
python-2.6.9-0 => not python-2.7.4-0
python-2.6.8-6 => not python-2.7.6-2
mklfft-2.0-np19py27_p0 => mkl-rt-11.1-p0
mklfft-2.0-np19py34_p0 => mkl-rt-11.1-p0
python-2.6.8-5 => not python-3.4.2-1
python-2.7.7-2 => not python-3.4.2-0
scipy-0.12.0-np16py27_p0 or scipy-0.12.0-np17py27_p0
python-2.6.8-6 => not python-3.4.1-1
python-2.7.3-7 => not python-3.4.1-2
python-2.7.7-1 => not python-3.4.2-0
python-2.6.9-0 => not python-3.4.2-1
python-2.6.8-5 => not python-2.7.8-0
python-2.7.10-0 => not python-3.4.1-2
python-3.4.0-0 or python-3.4.1-0 or python-3.4.1-1 or python-3.4.1-2 or python-3.4.2-0 or python-3.4.2-1 or python-3.4.3
-0
python-2.6.9-0 => not python-2.7.6-2
python-2.6.8-5 => not python-3.4.3-0
python-2.7.8-0 => not python-3.4.2-0
mklfft-1.0-np18py33_p0 => mkl-rt-11.1-p0
python-2.6.8-5 => not python-2.7.5-1
python-2.6.9-1 => not python-3.4.3-0
mklfft-1.0-np19py26_p0 => python-2.6.8-5 or python-2.6.8-6 or python-2.6.9-0 or python-2.6.9-1
python-2.6.8-6 => not python-3.4.1-0
scipy-0.12.0-np17py27_p0 => python-2.7.10-0 or python-2.7.3-2 or python-2.7.3-3 or python-2.7.3-4 or python-2.7.3-5 or p
ython-2.7.3-6 or python-2.7.3-7 or python-2.7.4-0 or python-2.7.5-0 or python-2.7.5-1 or python-2.7.5-2 or python-2.7.6-
0 or python-2.7.6-2 or python-2.7.7-0 or python-2.7.7-1 or python-2.7.7-2 or python-2.7.8-0 or python-2.7.9-0 or python-
2.7.9-1
python-2.7.10-0 => not python-3.4.2-0
python-2.6.9-0 => not python-3.4.1-1
python-2.6.8-6 => not python-2.7.5-2
mkl-rt-11.0-p2 => not mkl-rt-11.1-p0
mklfft-1.0-np18py27_p0 => mkl-rt-11.1-p0
python-2.6.9-1 => not python-3.4.1-0
python-2.6.8-6 => not python-2.7.9-1
python-2.7.3-3 => not python-3.4.2-0
python-2.6.9-1 => not python-2.7.9-1
python-2.6.8-5 => not python-2.7.5-2
python-2.6.8-6 => not python-3.4.0-0
python-2.6.9-1 => not python-2.7.5-2
python-2.6.9-0 => not python-3.4.1-2
python-2.6.8-5 => not python-2.7.9-1
mklfft-1.0-np19py27_p0 => mkl-rt-11.1-p0
python-2.7.3-2 => not python-3.4.2-0
mkl-rt-11.0-p1 => not mkl-rt-11.1-p0
python-2.6.8-5 => not python-3.4.0-0
mklfft-2.0-np19py33_p0 => mkl-rt-11.1-p0
python-2.7.3-5 => not python-3.4.2-0
python-2.6.9-0 => not python-2.7.5-2
python-2.6.8-5 => not python-2.7.3-4
python-2.7.7-0 => not python-3.4.2-0
python-2.6.9-1 => not python-3.4.0-0
python-2.7.3-4 => not python-3.4.2-0
python-2.6.9-0 => not python-2.7.9-1
python-2.6.8-5 => not python-2.7.6-0
python-2.7.7-1 => not python-3.4.1-2
python-2.7.9-0 => not python-3.4.2-0
python-2.6.9-0 => not python-3.4.0-0
python-2.7.3-2 => not python-3.4.1-2
mklfft-1.0-np18py26_p0 => python-2.6.8-5 or python-2.6.8-6 or python-2.6.9-0 or python-2.6.9-1
python-2.6.8-5 => not python-3.4.1-0
python-2.6.8-5 => not python-2.7.3-5
python-2.6.9-1 => not python-3.4.1-2
mkl-rt-11.0-p0 => not mkl-rt-11.1-p0
python-2.7.5-0 => not python-3.4.1-2
mklfft-1.0-np18py34_p0 => mkl-rt-11.1-p0
python-2.6.8-5 => not python-2.7.6-2
python-2.6.8-6 => not python-3.4.3-0
mklfft-2.0-np19py26_p0 => mkl-rt-11.1-p0
python-2.6.9-1 => not python-2.7.6-2
python-2.6.8-6 => not python-2.7.4-0
python-2.7.9-0 => not python-3.4.1-2
python-2.6.9-0 => not python-3.4.1-0
python-2.6.9-1 => not python-2.7.4-0
python-2.7.5-0 => not python-3.4.2-0
python-2.6.8-5 => not python-3.4.1-1
mklfft-1.0-np18py26_p0 or mklfft-1.0-np18py27_p0 or mklfft-1.0-np18py33_p0 or mklfft-1.0-np18py34_p0 or mklfft-1.0-np19p
y26_p0 or mklfft-1.0-np19py27_p0 or mklfft-1.0-np19py33_p0 or mklfft-1.0-np19py34_p0 or mklfft-2.0-np19py26_p0 or mklfft
-2.0-np19py27_p0 or mklfft-2.0-np19py33_p0 or mklfft-2.0-np19py34_p0
python-2.7.3-3 => not python-3.4.1-2
python-2.7.7-2 => not python-3.4.1-2
python-2.6.8-5 => not python-2.7.3-6
scipy-0.12.0-np17py27_p0 => mkl-rt-11.0-p0 or mkl-rt-11.0-p1 or mkl-rt-11.0-p2
python-2.6.9-1 => not python-3.4.1-1
mklfft-1.0-np19py33_p0 => mkl-rt-11.1-p0
python-2.7.3-7 => not python-3.4.2-0
python-2.6.9-0 => not python-3.4.3-0
python-2.6.8-5 => not python-2.7.7-0
python-2.6.8-6 => not python-3.4.2-1
python-2.7.6-0 => not python-3.4.2-0
```

Why is the `update --all` failing?
Thank you!

**Comment by asmeurer** [comment]
What is the regular hint that it gives you?

**Comment by bryanwweber** [comment]
It varies every time I run the command, but several are:

---
    - odo
    - scikit-learn
---
    - pytables
    - scipy
    - accelerate
    - python 3.4*
---
    - scikit-image
    - accelerate
    - pytables
    - python 3.4*
---
    - pytables
    - odo
    - pep8
    - scipy

Please let me know any further information you need. Thanks for helping!

**Comment by asmeurer** [comment]
I see. I think there is a bug in the hint code, which isn't including the fact that these only conflict with `mkl`. 

The problem it seems is that there is an old scipy package on Windows with the mkl feature. The package is so old that it is only built against numpy 1.6 and 1.7 and Python 2.7, whereas packages like odo are only built against numpy 1.9. 

I'll have to think about how to fix it. For now, you'll have to update packages individually.

**Comment by bryanwweber** [comment]
OK thanks! This just started happening today, so I'm not sure what would have changed. Is there a way to build scipy with MKL against Python 3.4 and the newest numpy? Or rather, I'm sure there's a way, but do you know of any good instructions?

**Comment by asmeurer** [comment]
The thing is that on Windows, the default scipy already is built against mkl, which is why we stopped updating the package with the mkl feature. Perhaps a workaround would be to take the existing scipy, add the mkl feature to it, and upload that.

**Comment by bryanwweber** [comment]
Is there an easy way for a user to do that, or is that something that Continuum has to handle? I have done no more than use `conda build` following the tutorial, so I'm not sure where to start for this.

**Comment by asmeurer** [comment]
It would be doable. It would consist of unpacking the package, changing some metadata, and repacking it. It's really something that we need to fix on our end, though.

**Comment by bryanwweber** [comment]
OK I will let you handle it then and wait patiently for the update. Thank you for all your help! Have a great day!

**Comment by kalefranz** [comment]
Anaconda 2.2 shipped with conda 3.10.0, so I'm assuming that's what this issue is in reference to.  Conda's solver had a lot of work done with the conda 4.0 and 4.1 releases.  If there's an ongoing issue here, feel free to file a new ticket after validating the problem against recent versions of conda (latest is conda 4.3.17).
